cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs

```diff
@@ -1012,35 +1015,78 @@ runTransactionCalculateMinFeeCmd
+  unwitnessed <-
+    firstExceptT TxCmdCddlError . newExceptT
+      $ readFileTxBody txbodyFile
+  pparams <-
+    firstExceptT TxCmdProtocolParamsError
+      $ readProtocolParameters protocolParamsFile
+
+  let nShelleyKeyWitW32 = fromIntegral nShelleyKeyWitnesses
+
+  shelleyfee <- case unwitnessed of
+    IncompleteCddlFormattedTx (InAnyShelleyBasedEra sbe unwitTx) -> do
       let txbody =  getTxBody unwitTx
-      ledgerPparams <-
-        firstExceptT TxCmdProtocolParamsConverstionError $
-        hoistEither $ toLedgerPParams sbe pparams
-      let Lovelace fee = evaluateTransactionFee sbe
-                            ledgerPparams
-                            txbody
-                            (fromIntegral nShelleyKeyWitnesses)
-                            (fromIntegral nByronKeyWitnesses)
-
-      liftIO $ putStrLn $ (show fee :: String) <> " Lovelace"
-
-    UnwitnessedCliFormattedTxBody anyTxBody -> do
-      InAnyShelleyBasedEra sbe txbody <- pure anyTxBody
-      ledgerPparams <-
-        firstExceptT TxCmdProtocolParamsConverstionError $
-        hoistEither $ toLedgerPParams sbe pparams
-      let Lovelace fee = evaluateTransactionFee sbe
-                            ledgerPparams
-                            txbody
-                            (fromIntegral nShelleyKeyWitnesses)
-                            (fromIntegral nByronKeyWitnesses)
-
-      liftIO $ putStrLn $ (show fee :: String) <> " Lovelace"
+      lpparams <- getLedgerPParams sbe pparams
+      pure $ evaluateTransactionFee sbe lpparams txbody nShelleyKeyWitW32 0
+    UnwitnessedCliFormattedTxBody (InAnyShelleyBasedEra sbe txbody) -> do
+      lpparams <- getLedgerPParams sbe pparams
+      pure $ evaluateTransactionFee sbe lpparams txbody nShelleyKeyWitW32 0
+
+  let byronfee = calculateByronWitnessFees
+                   (protocolParamTxFeePerByte pparams)
+                   nByronKeyWitnesses
+
+  let Lovelace fee = shelleyfee + byronfee
+
+  liftIO $ putStrLn $ (show fee :: String) <> " Lovelace"
+
+getLedgerPParams :: forall era. ()
+  => ShelleyBasedEra era
+  -> ProtocolParameters
+  -> ExceptT TxCmdError IO (Ledger.PParams (ShelleyLedgerEra era))
+getLedgerPParams sbe pparams =
+  firstExceptT TxCmdProtocolParamsConverstionError $
+  hoistEither $ toLedgerPParams sbe pparams
+
+-- Extra logic to handle byron witnesses.
+-- TODO: move this to Cardano.API.Fee.evaluateTransactionFee.
+calculateByronWitnessFees :: ()
+  => Lovelace -- ^ The tx fee per byte (from protocol parameters)
+  -> Int      -- ^ The number of Byron key witnesses
+  -> Lovelace
+calculateByronWitnessFees txFeePerByte byronwitcount =
+    Lovelace
+      $ toInteger txFeePerByte
+      * toInteger byronwitcount
+      * toInteger sizeByronKeyWitnesses
+  where
+    sizeByronKeyWitnesses = smallArray + keyObj + sigObj + ccodeObj + attrsObj
+
+    smallArray  = 1
+
+    keyObj      = 2 + keyLen
+    keyLen      = 32
+
+    sigObj      = 2 + sigLen
+    sigLen      = 64
```

Comment:
Some of these appear to be magic numbers. Again not critical for the PR but it would be nice to call the appropriate `encodedSizeExpr` from `cardano-ledger-binary` for these numbers.

Reply:
These magical numbers are copied directly from Cardano.API.estimateTransactionFee, which is what Cardano CLI calculate-min-fee was using previously.

https://github.com/IntersectMBO/cardano-api/blob/109612cab4173442c1230b7ebbb869076ac04d7c/cardano-api/internal/Cardano/Api/Fees.hs#L157

Reply:
The magic numbers won't be needed if/when @lehins PR to Cardano Ledger gets merged.